### ISSUE TITLE
useRxCollection does not support lazy collection initialization

### ISSUE THREAD

**Issue by plmercereau** [cvara/rxdb-hooks#27]
Hello,
First of all, a big thanks for this awesome library!

I am loading collections after DB creation with `db.addCollections`. The problem (I think) is the collection is not created yet when the `useRxCollection` hook is called.
As a result, the above code:
```js
  const { isFetching, result } = useRxData(
    name,
    (collection) => collection.find()
  )
```
ends up with `isFetching` being `false`, and
```js
const collection = useRxCollection(name)
```
would also send an error if the collection is not created yet.
Is there a way to make `rxdb-hooks` work in this use case?
Many thanks in advance!

**Comment by cvara** [comment]
Hi @plmercereau.

Lazily generating collections is indeed not handled at the moment, but seems like a legit use case.

AFAIK rxdb does not allow db consumers to observe for new collections being added via the [db.$ api](https://rxdb.info/rx-database.html#observe-with-) (tested on v8.x, will need to confirm for 9), so another solution needs to be devised so that ultimately `useRxCollection` gets notified, picks up the new collection and returns it.

I'll think on it some more once I get the chance. In the meantime I've put together a (failing) test case for this in #28.

**Comment by plmercereau** [comment]
Yes, @pubkey mentioned [on gitter](https://gitter.im/pubkey/rxdb?at=60aabf93850bfa2d3bdce0ca) he was not really fond of the idea of coding something like `db.collextions$` in rxbd core, which could have been helpful to subscribe to lazy collection loading.
I guess his suggestion to add a plugin may be an option, I'll sleep on it and will try something that would pass your test..

**Comment by plmercereau** [comment]
I created a mini-plugin that would implement `db.collections$` but I couldn't find a way to make it pass your test - I guess something needs to be adjusted on the React side, or there's a problem with this design...

**Comment by cvara** [comment]
Thanks for taking the time to try this out @plmercereau.

The reason the tests are failing (#29) is that at the moment rxdb-hooks use version 8 of rxdb for the tests (the api itself is still compatible with rxdb 9). If the plugin path is to be followed we would lose backwards compatibility with rxdb 8, which isn't necessarily a bad thing. I will bump rxdb to 9 and retry based on your approach.

**Comment by cvara** [comment]
I've put together a working prototype that makes the test pass. Still need to improve typings & coverage.

**Comment by cvara** [comment]
Just merged #28 that resolves this. Thanks for your input @plmercereau!

Will push a new major release within the next few days once I properly update the readme.

**Comment by plmercereau** [comment]
Great! You may have seen that the [next rxdb refactoring may improve the way it would work with react hooks](https://github.com/pubkey/rxdb/blob/master/orga/before-next-major.md) - maybe an opportunity to contribute and to make things even more sustainable?
Many thanks for solving this up!
